### ISSUE TITLE
Maya installed, but css2rss throws weird error

### ISSUE THREAD

**Issue by seventyiris83** [Owyn/CSS2RSS#12]
hi, i had to wipe Windows clean the other day and as a consequence had to reinstall python (3.13.150.0), beautifulsoup and maya for css2rss to fully work again.

i've got a weird bug here where the articles of the feeds using css2rss get fetched in the form of an error, see the screenshot:

![css2rss-bug](https://github.com/user-attachments/assets/efa7e65c-fd4b-4fc0-9192-367906221136)

when running `pip install maya` i was getting an error telling me it could't get past the step "Building wheel for pendulum (pyproject.toml)", so i [downloaded](https://static.rust-lang.org/rustup/dist/i686-pc-windows-gnu/rustup-init.exe) rust and installed it, then ran `rustup toolchain install stable-x86_64-pc-windows-gnu` followed by `rustup default stable-x86_64-pc-windows-gnu`. 

I retried `pip install maya` but then rust was having an error telling me i was missing gcc, so i also got that [here](https://github.com/niXman/mingw-builds-binaries), added gcc to PATH, and was able to finally install maya. but it doesn't work as i'm getting the error mentioned above. 

any idea what went wrong here and how i can get maya properly working again? thank you!

**Comment by Owyn** [comment]
That doesn't look an error, that looks like a working output tho 🤔 

can you post the full error? by running rssguard in debug mode for example - it should write errors into its logfile (i think) - https://rssguard.readthedocs.io/en/stable/contrib/bugs.html 

> pip install maya 

you should try that from admin cmd

> rustup 

it is written that it's for installing Rust and that's it 🤔 

I was having problems with reinstalling Python manually, and nothing really worked, only Chocolatey packet manager managed to install it correctly for me, that might also help you

Anyway, first of all - try removing the last argument `"span.flex"` from css2rss (selector for Time) and see if it works fine when Maya is not activated to grab and parse time

**Comment by seventyiris83** [comment]
thanks for the quick response

> That doesn't look an error, that looks like a working output tho 🤔

yeah, i just call it an error as RSS Guard returns "Script failed: script threw an error" :smile:

> can you post the full error? by running rssguard in debug mode for example - it should write errors into its logfile (i think) - https://rssguard.readthedocs.io/en/stable/contrib/bugs.html

sure: 

> > pip install maya
> 
> you should try that from admin cmd

tried this, all requirements are already satisfied

> > rustup
> 
> it is written that it's for installing Rust and that's it 🤔

i used that in the past when i couldn't install maya a while ago (before fresh windows install), it could't get past the step "Building wheel for pendulum (pyproject.toml)", using rustup allowed it to go ahead with what it needed to install. Maya worked fine then but not this time

> I was having problems with reinstalling Python manually, and nothing really worked, only Chocolatey packet manager managed to install it correctly for me, that might also help you

thanks for the suggestion. i think python is installed correctly, when i run python --version it returns 3.13.0

> Anyway, first of all - try removing the last argument `"span.flex"` from css2rss (selector for Time) and see if it works fine when Maya is not activated to grab and parse time

yes, css2rss works fine when Maya is not activated

**Comment by Owyn** [comment]
Try changing `ValueError` (line 76) to `BaseException` in CSS2RSS.py file and see if it starts working

**Comment by seventyiris83** [comment]
> Try changing ValueError (line 76) to BaseException in CSS2RSS.py file and see if it starts working

same result unfortunately

**Comment by Owyn** [comment]
and if you change both (L73 & L76) `except BaseException:` to `except:` ?

**Comment by seventyiris83** [comment]
> and if you change both (L73 & L76) except BaseException: to except: ?

same

**Comment by Owyn** [comment]
That should mean it errors when importing the Maya module 🤔 

Here, I have added a more-user friendly error display when that happens
![image](https://github.com/user-attachments/assets/45ba3c3b-7e5d-4353-837e-e2e0eb90d536)

try the new css2rss.py version ( https://github.com/Owyn/CSS2RSS/raw/refs/heads/main/css2rss.py ) and see if it prints the error better now

**Comment by seventyiris83** [comment]
wiped all maya install files in python's site-packages folder: the error message you added does display (no module named 'maya')

re-installed maya through admin cmd, i'm right back to the same result of this ticket

**Comment by Owyn** [comment]
![pain](https://github.com/user-attachments/assets/a854a174-f428-40d7-af77-f078613e4f72)

Anyway, here's a script's version which ignores all errors and never errors (does not mean it will work tho): 
[css2rss.zip](https://github.com/user-attachments/files/17893985/css2rss.zip)

**Comment by seventyiris83** [comment]
> ￼
> 
> Anyway, here's a script's version which ignores all errors and never errors (does not mean it will work tho):
> css2rss.zip

no luck. I'll go through the whole python install again and avoid the rust-up install altogether, that's my #1 suspect causing this

**Comment by Owyn** [comment]
> no luck.

that literally means it's not this script's fault since erroring is disabled in it fully

**Comment by seventyiris83** [comment]
> that literally means it's not this script's fault since erroring is disabled in it fully

yeah, this isn't css2rss's fault. RSS Guard does say when the script isn't present in the data folder so i don't understand why i get that same error output even when trying your error-disabled version of it

wiped python, everything rust-up installer installed, bs4 and maya then re-installed python, bs4 and maya: same result.

Tested different versions of RSS Guard, same result

Are you on the latest version of python 3.13?

**Comment by Owyn** [comment]
```
C:\Users\owyn>python -V
Python 3.12.4

C:\Users\owyn>pip list
Package         Version
--------------- -----------
beautifulsoup4  4.12.3
dateparser      1.2.0
humanize        4.10.0
maya            0.6.1
pendulum        3.0.0
pip             24.3.1
PyQt5           5.15.10
PyQt5-Qt5       5.15.2
PyQt5_sip       12.15.0
PyQt6           6.7.1
PyQt6-Qt6       6.7.3
PyQt6_sip       13.8.0
python-dateutil 2.9.0.post0
pytz            2024.1
regex           2024.7.24
six             1.16.0
snaptime        0.2.4
soupsieve       2.5
time-machine    2.14.2
tinycss2        1.4.0
tzdata          2024.1
tzlocal         5.2
webencodings    0.5.1
webviewpy       1.4.0
```

3.12.4 is the latest one chocolatey offers so I'm using it
